### PR TITLE
Only run the install task on the code-quality-reports codebase

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

File: eng/pipelines/client.yml [hunk 1]
```diff
@@ -72,13 +72,13 @@ jobs:
         condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'))
         displayName: 'Component Detection'
 
-      # We `install` separately from running `site:site site:stage` so that the `install` brings in the non-shipping-modules,
-      # but we don't include them in the Maven site commands (so that we don't generate reports for the non-shipping modules).
+      # We `install` the code quality reports tooling into our local m2 cache separately from building the Maven project
+      # reports. This means it is available as part of that, but also so that this is not documented in the proejct report.
```
Comment (conniey) [inline eng/pipelines/client.yml]: nit: proejct -> project

Comment (JonathanGiles) [reply eng/pipelines/client.yml]: You and your words!

**Comment by JonathanGiles** [comment]
whomp whomp. Some build artifacts aren't generated like this. I'll tweak and update PR.

**Comment by JonathanGiles** [comment]
> Looks reasonable to me. I can replicate this in the analyze step I'm using (eng/pipelines/templates/jobs/archetype-sdk-client.yml.

If this PR builds successfully, I'll do the replication before I merge.

**Comment by JonathanGiles** [comment]
@mitchdenny @weshaggard As a proof of concept, this PR takes the Analyze job from around 10:52 to 5:24, so roughly reduced by half. The way I mainly achieved this was by not producing HTML artifacts (e.g. Maven Site, JavaDoc, Jacoco, spotbugs reports or checkstyle reports). What I would like to see is these generated daily, rather than per-PR. Is this something that we could do (in which case, much of what I have commented out in this PR would move into that job instead)

**Comment by JonathanGiles** [comment]
@mitchdenny @weshaggard As a proof of concept, this PR takes the Analyze job from around 10:52 to 5:24, so roughly reduced by half. The way I mainly achieved this was by not producing HTML artifacts (e.g. Maven Site, JavaDoc, Jacoco, spotbugs reports or checkstyle reports). What I would like to see is these generated daily, rather than per-PR. Is this something that we could do (in which case, much of what I have commented out in this PR would move into that job instead)

**Comment by danieljurek** [comment]
The `client.yml` file defines the nightly run. To shorten the inner-loop cycle we should remove steps from `archetype-sdk-client.yml`.

**Comment by mitchdenny** [comment]
We should cut client.yml to use the archetype template anyway. I'll do that soon! But these changes look good.

**Comment by JonathanGiles** [comment]
Build failed due to networking failure.

**Comment by JonathanGiles** [comment]
@weshaggard What's the best way here to get this merged? Due to the network failure I either need to know the right command to re-run the build, or have it force-merged?

**Comment by weshaggard** [comment]
@jonathanGiles You should be able to re-run any failed leg via the Checks tab on github. Assuming that doesn't work then the simply copy the job name listed in the details and call `/azp run <job>`. As long as we can get the `/azp run azure-sdk-for-java - client` to green then I'm good with force merging.

**Comment by JonathanGiles** [comment]
> @JonathanGiles You should be able to re-run any failed leg via the Checks tab on github. Assuming that doesn't work then the simply copy the job name listed in the details and call `/azp run <job>`. As long as we can get the `/azp run azure-sdk-for-java - client` to green then I'm good with force merging.

Yeah - the 're-run' links don't seem to work for me. I'll re-run it as outlined now.

**Comment by weshaggard** [comment]
>the 're-run' links don't seem to work for me.

I've found that the re-run links don't actually work until all the jobs in the particular pipeline has finished. If any of them are still running then it doesn't work. The DevOps team should really work on improving that entire experience though because it doesn't give people any feedback.

**Pull request merged**